routers/web/repo/commit.go

```diff
@@ -458,3 +464,54 @@ func processGitCommits(ctx *context.Context, gitCommits []*git.Commit) ([]*git_m
 	}
 	return commits, nil
 }
+
+// GroupedCommits defines the structure for grouped commits.
+type GroupedCommits struct {
+	Date    timeutil.TimeStamp
+	Commits []*git_model.SignCommitWithStatuses
+}
+
+// GroupCommitsByDate groups the commits by date (in days).
+func GroupCommitsByDate(commits []*git_model.SignCommitWithStatuses) []GroupedCommits {
+	// Use Unix timestamp of date as key (truncated to day)
+	grouped := make(map[int64][]*git_model.SignCommitWithStatuses)
+
+	for _, commit := range commits {
+		var sigTime time.Time
+		if commit.Committer != nil {
+			sigTime = commit.Committer.When
+		} else if commit.Author != nil {
+			sigTime = commit.Author.When
+		}
+
+		// Truncate time to date part (remove hours, minutes, seconds)
+		year, month, day := sigTime.Date()
+		dateOnly := time.Date(year, month, day, 0, 0, 0, 0, sigTime.Location())
+		dateUnix := dateOnly.Unix()
+
+		grouped[dateUnix] = append(grouped[dateUnix], commit)
+	}
+
+	// Create result slice with pre-allocated capacity
+	result := make([]GroupedCommits, 0, len(grouped))
+
+	// Collect all dates and sort them
+	dates := make([]int64, 0, len(grouped))
+	for dateUnix := range grouped {
+		dates = append(dates, dateUnix)
+	}
+	// Sort dates in descending order (most recent first)
+	sort.Slice(dates, func(i, j int) bool {
+		return dates[i] > dates[j]
+	})
```

Comment:
Another edge case: git commit's date could be manually set.

So you could see a commit list like this: 

```
commit4 2025-04-10
commit3 2025-04-01
commit2 2025-04-30
commit1 2025-04-15
```

And the graph is commit1->commit2->commit3->commit4.

What's the expected output for such commit list? It needs some tests to clarify the behavior.

Reply:
The to be discussed here is which time field to use. Currently, the list uses the committer's time both before and after grouping. If adjustment is needed, we can make modifications later.